src/DataGrid.tsx

```diff
@@ -880,57 +872,93 @@ function DataGrid<R, SR, K extends Key>(
+      onFill == null
+    ) {
+      return;
     }
 
-    return {
-      selectedCellIdx: selectedPosition.idx,
-      onKeyDown: handleKeyDown,
-      selectedCellDragHandle:
-        // Cell drag is not supported on a treegrid
-        !hasGroups && onFill != null ? (
-          <DragHandle
-            rows={rawRows}
-            columns={columns}
-            selectedPosition={selectedPosition}
-            isCellEditable={isCellEditable}
-            latestDraggedOverRowIdx={latestDraggedOverRowIdx}
-            onRowsChange={onRowsChange}
-            onFill={onFill}
-            setDragging={setDragging}
-            setDraggedOverRowIdx={setDraggedOverRowIdx}
-          />
-        ) : undefined
-    };
+    return (
+      <DragHandle
+        rows={rawRows}
+        columns={columns}
+        selectedPosition={selectedPosition}
+        isCellEditable={isCellEditable}
+        latestDraggedOverRowIdx={latestDraggedOverRowIdx}
+        onRowsChange={onRowsChange}
+        onFill={onFill}
+        setDragging={setDragging}
+        setDraggedOverRowIdx={setDraggedOverRowIdx}
+      />
+    );
+  }
+
+  function getCellEditor(rowIdx: number) {
+    if (selectedPosition.rowIdx !== rowIdx || selectedPosition.mode === 'SELECT') return;
+
+    const { idx, row } = selectedPosition;
+    const column = columns[idx];
+    const colSpan = getColSpan(column, lastFrozenColumnIndex, { type: 'ROW', row });
+
+    return (
+      <EditCell
+        key={column.key}
+        column={column}
+        colSpan={colSpan}
+        row={row}
+        editorPortalTarget={editorPortalTarget}
+        onKeyDown={handleKeyDown}
+        onRowChange={handleEditorRowChange}
+        onClose={handleOnClose}
+      />
+    );
   }
 
   function getViewportRows() {
     const rowElements = [];
     let startRowIndex = 0;
-    for (let rowIdx = rowOverscanStartIdx; rowIdx <= rowOverscanEndIdx; rowIdx++) {
+
+    const { idx: selectedIdx, rowIdx: selectedRowIdx } = selectedPosition;
+    const cellIsWithinBounds = isCellWithinBounds(selectedPosition);
+    const startRowIdx =
+      cellIsWithinBounds && selectedRowIdx < rowOverscanStartIdx
+        ? rowOverscanStartIdx - 1
+        : rowOverscanStartIdx;
+    const endRowIdx =
+      cellIsWithinBounds && selectedRowIdx > rowOverscanEndIdx
+        ? rowOverscanEndIdx + 1
+        : rowOverscanEndIdx;
+
+    for (let viewportRowIdx = startRowIdx; viewportRowIdx <= endRowIdx; viewportRowIdx++) {
+      const isRowOutsideViewport =
+        viewportRowIdx === rowOverscanStartIdx - 1 || viewportRowIdx === rowOverscanEndIdx + 1;
+      const rowIdx = isRowOutsideViewport ? selectedRowIdx : viewportRowIdx;
+
+      let rowColumns = viewportColumns;
+      const selectedColumn = columns[selectedIdx];
+      // selectedIdx can be -1 if grouping is enabled
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+      if (selectedColumn !== undefined) {
```

Comment:
Would this work?
```suggestion
      if (cellIsWithinBounds) {
```

Reply:
No, because `cellIsWithinBounds` is true if `groupBy.length > 0 && idx === -1`